hack/tools/internal/tilt-prepare/main.go

```diff
@@ -848,6 +851,16 @@ func prepareWorkload(name, prefix, binaryName, containerName string, objs []unst
 	})
 }
 
+func containerHasMetricsPort(ports []corev1.ContainerPort) bool {
+	for _, port := range ports {
+		if port.ContainerPort == 8080 && port.Name == "metrics" && port.Protocol == corev1.ProtocolTCP {
```

Comment:
```suggestion
		if port.Name == "metrics" {
```
With the current check we would still run into errors right?

Reply:
Or would we want to update existing metrics ports to 8080? (so it makes sense together with l.831?)

Reply:
Btw I was planning to get rid of this entire section with #9289

Reply:
If we would get rid of it with #9289 I'm fine with a pragmatic stopgap solution to unblock you in the meantime

Reply:
TBH I would avoid at all adding the `metrics` port, since if the addon is not exposing it there's nothing to scrape from.

Reply:
Yup. That is my plan for #9289. I would just like to preserve the functionality until then (should take 1-2 weeks for me to get #9289 in shape)

Reply:
The reason why we currently have it is because per default we don't want to expose an insecure metrics port, but for development we would like to be able to work with metrics

Reply:
> So maybe for now let's go with `if port.Name == "metrics" {`?

Thanks, force pushed with be683738a

Reply:
Ah ups, the port has to be unique not the name :) My bad